packages/application/src/shell.ts

```diff
@@ -972,22 +971,25 @@ namespace Private {
   }
 
   /**
-   * A message hook that adds and removes the .jp-Activity class to widgets in the dock panel.
+   * A factory returning a message hook for main area dock panel bookkeeping.
    */
   export
-  function activityClassHook(handler: IMessageHandler, msg: Message): boolean {
-    switch (msg.type) {
-      case 'child-added':
-        (msg as Widget.ChildMessage).child.addClass(ACTIVITY_CLASS);
-        break;
-      case 'child-removed':
-        (msg as Widget.ChildMessage).child.removeClass(ACTIVITY_CLASS);
-        break;
-      default:
-        break;
+  function dockChildHookFactory(tracker: FocusTracker<Widget>) {
+    return (handler: IMessageHandler, msg: Message): boolean => {
```

Comment:
By creating an anonymous function hook, you'll never be able to uninstall it unless you clear all message data for the target.

Reply:
Good point. Problem is that we need access to the focus tracker.

Reply:
Why not make it a bound method on the class?   `this._dockChildHook = () => {...}`.

Reply:
(updated syntax above)

Reply:
And it's not a problem in this case - the dock panel and the focus tracker lives as long as the application shell does. The dock panel dispose should be called in the shell dispose to clear this out, but it wasn't before, presumably because the shell lives as long as the page does.

Reply:
No, it's more nuanced than that. The dock panel is added to the shell widget's layout, so it is appropriately disposed when the shell widget is disposed. So I think we're all good as far as cleanup goes.

Reply:
> Why not make it a bound method on the class? private _dockChildHook = () => {...}.

That's cleaner. Thanks.